dotnet/src/Agents/Core/ChatCompletionAgent.cs

```diff
@@ -78,7 +78,11 @@ public override async IAsyncEnumerable<AgentResponseItem<ChatMessageContent>> In
         // Get the context contributions from the AIContextProviders.
 #pragma warning disable SKEXP0110 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
         var providersContext = await chatHistoryAgentThread.AIContextProviders.ModelInvokingAsync(messages, cancellationToken).ConfigureAwait(false);
-        kernel.Plugins.AddFromAIContext(providersContext, "Tools");
+        if (providersContext.AIFunctions is { Count: > 0 })
+        {
+            //TODO: Add a plugin name collision handling mechanism to address cases when a plugin with the name "Tools" is already registered in the kernel.
```

Comment:
Suggest creating the issue and removing the comment.

```suggestion
            //TODO: Add a plugin name collision handling mechanism to address cases when a plugin with the name "Tools" is already registered in the kernel.
```

Reply:
https://github.com/microsoft/semantic-kernel/issues/12320